src/views/components/SharePage/index.jsx

```diff
@@ -0,0 +1,92 @@
+import { useRef, useCallback } from 'react'
+import { useNavigate } from "react-router-dom";
+import Typography from '@mui/material/Typography'
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import { RESPONSIVE_PANEL_SPACING } from "../ResponsivePanel";
+import ROUTE_NAMES from "../../../constants/routeNames";
+
+import {
+  WhatsappShareButton,
+  WhatsappIcon,
+  EmailShareButton,
+  EmailIcon,
+} from "react-share";
+
+const { HOME } = ROUTE_NAMES;
+
+const SharePage = ({ subcribe }) => {
+  
+  const normal_share =useRef(null);
+
+  const navigate = useNavigate();
+  const button1 = useRef(null);
+  const button2 = useRef(null);
+  const navigateToHome = useCallback(() => navigate(HOME), [navigate]);
+  const share_click = () => {
+    // e.preventDefault();
+    if (navigator.share) {
+      navigator
+        .share({
+          title: "Lets help kids to play safer!",
+          url: "https://kidsplaysafer.sg",
+        })
+        .catch(console.error);
+    } else {
+      //fallback if the normal_share api is not supported by user's browser
+      button1.current.style.display = "inline-flex";
+      button2.current.style.display = "inline-flex";
+      button1.current.style.zIndex = "2";
+      button2.current.style.zIndex = "2";
+      normal_share.current.style.display ="none";
+    }}
+
+  return (
+    <>
+      <Box mb={RESPONSIVE_PANEL_SPACING}>
+        {subcribe === 1 &&
+        <Typography variant="h4">Thank You for subscribing!</Typography>}
+        {subcribe === 2 &&
+        <Typography variant="h4">Play Again?</Typography>}
+      </Box> 
+
+      <Box mb={RESPONSIVE_PANEL_SPACING}>
+
+      <Button
+        onClick={navigateToHome}
+        variant='outlined'
+        sx={{ margin: '10px' }}
+      >
+        Return to home
+      </Button>
+
+      <Button
+        onClick={share_click}
+        variant='outlined'
+        sx={{ margin: '10px' }}
+        ref ={normal_share}
+      >
+        Share
+      </Button>
```

Comment:
You might be able to simplify this code a bit by doing the following:
```
{ !!navigator.share
  ? <Button
        onClick={handleShare}
        variant='outlined'
        sx={{ margin: '10px' }}
      >
        Share
      </Button>
  : <>
    <WhatsappShareButton url="https://kidsplaysafer.sg" marginRight:"20px"}}>
        <WhatsappIcon size={50} />
      </WhatsappShareButton>
      <EmailShareButton
        url="https://kidsplaysafer.sg"
        subject="Lets help kids to play safer!"
        body=" Enjoy and learn more online safety with your child/children while do this quiz with them!"
      >
        <EmailIcon size={50} round={true} />
      </EmailShareButton>
  </>
}
```

Then you can remove the 3 `useRef` and simplify the `share_click` function as follow:
```
const handleShare = useCallback(() => {
    navigator
      .share({
        title: "Lets help kids to play safer!",
        url: "https://kidsplaysafer.sg",
      })
      .catch(console.error);
  }. [navigator, navigator.share]); // Double check this, I'm not sure it will work :p
```

What do you think? I'm writing the code directly in the comment so the syntax might be slightly incorrect but you get the idea :)

Reply:
The reason I'm recommending this is so you keep the logic to decide what needs to be displayed in the jsx, and you don't need to rely on `display: none` and CSS anymore to handle the fallback, instead the ternary `{ !!navigator.share ? <.../> : <.../> }` will do that for you.